### DISCUSSION TITLE
Form is only repopulated with field data after submission if values have changed

### DISCUSSION THREAD

**Discussion by paulchoix** [sveltejs/kit#8886] [Q&A]
I'm having trouble with a form that is supposed to keep the input field values after submission. Essentially, the form is for modifying user data: it is generated with the current user data and upon submission the fields should stay the same (since they contain the same values that are sent to the server). However, when I submit the form, all the fields are emptied _except_ the ones that have been modified.

For example, the form is populated like this:

- Email: email@email.com
- First Name: John
- Last Name: Miller

If I change John to Joe and submit, the form is repopulated like this after submission:

- Email: 
- First Name: Joe
- Last Name

The `+page.svelte` file looks like this (I use components for my Form and Input fields):

```
<script>
	...
	/** @type {import('./$types').PageData} */
	export let data;
	let { _user } = data;

	...
	function handleUpdate({ detail }) {
		if (detail.status == 200) {
			_user = detail.data._user;
		}
	}
</script>

<FormPost name="profile" on:post_result={handleUpdate}>
	<Input name="email" type="email" label="Email" value={_user.email} required />
	<Input name="first_name" label="First Name" value={_user.first_name} required />
	<Input name="last_name" label="Last Name" value={_user.last_name} required />
	<Button>Submit</Button>
</FormPost>
```

In the `handleUpdate` function, the `_user` variable is correctly updated and contains all the fields when I log it to the console after the assignment. I don't think the problem is coming from my Input component since I have the same issue using HTML elements. My FormPost component uses custom enhancement like so:

```
use:enhance={() => {
	return async ({ result, update }) => {
	if (do_update) update();
		handleResult(result);
	};
}}
```

I'm at a loss, why is the form only repopulated when the values are different than the ones taken from the PageData?

**Comment by geoffrich** [comment]
This sounds similar to https://github.com/sveltejs/kit/issues/8513, though that issue populates the inputs using the `form` prop.
